src/plus/gk/account/subscriptionService.ts

```diff
@@ -488,6 +491,37 @@ export class SubscriptionService implements Disposable {
 		});
 	}
 
+	private requestExceptionCount = 0;
+	private requestsAreBlocked = false;
+
+	resetRequestExceptionCount(): void {
+		this.requestExceptionCount = 0;
+		this.requestsAreBlocked = false;
+	}
+
+	trackRequestException(): void {
+		this.requestExceptionCount++;
+
+		if (this.requestExceptionCount >= 5 && this.getAuthenticationSession() !== null) {
+			if (!this.requestsAreBlocked) {
+				void showIntegrationDisconnectedTooManyFailedRequestsWarningMessage('GitKraken');
+			}
+			this.requestsAreBlocked = true;
+			this.requestExceptionCount = 0;
+		}
+	}
+
+	private fetchApi(
+		path: string,
+		init?: RequestInit,
+		options?: { token?: string; organizationId?: string },
+	): Promise<Response> {
+		if (this.requestsAreBlocked) {
+			throw new Error('Requests are blocked');
```

Comment:
We're throwing an error now in places where we weren't before. Should we be catching/handling this from the callers? And if not, have we verified that it doesn't break anything?

Reply:
`fetchApi` is private inside of SubscriptionService and in all 3 places where it's called all exceptions are caught now.

But, we probably can handle it differently from other types of exceptions. For example, when `checkInAndValidateCore` catches and exception it changes subscription status. Maybe we shouldn't do it for this type of error?

Reply:
That's probably fine? Before these changes, it would hit this anyway if the fetch threw an error, so really the only new error we're introducing is from too many failed requests, in which case, we would probably want to hit that debugger and know about it anyway.